### PR TITLE
Adds an updateable profile to My Profile

### PR DESCRIPTION
**Pull request by ryaholm**
Modifies ExtendedUser to include fields that prepare it for the forum- avatar, bio, display name. Display name is something I am debating about. The idea is that it is like Twitter, where you have a handle but you can put whatever name you want to be your display name which is different from your @username: "Ryan Holm" (@c00ld00d). If this is too much, I can nix it. I'm deciding to do this "display name" instead of having first_name and last_name.

Modifies user_detail.html to have a django-bootstrap3 field where the values can be modified, such as a new avatar can be uploaded to avatar/{{user.id}}/{{avatar_file_name}}.

### CODE REVIEW

File: project_share/views.py [hunk 1]
```diff
@@ -402,3 +408,74 @@ def dispatch(self, request, *args, **kwargs):
...
+    form_class = forms.ProfileForm
+    model = User
+
+    def get_object(self, queryset=None):
+        return self.request.user
+
+    def form_valid(self, form):
+        return HttpResponseRedirect(reverse('home'))
+
+    def get_initial(self):
```
Comment (SillyInventor) [inline project_share/views.py]: Should include race, gender, and age if you're going to include them in the editables

File: project_share/views.py [hunk 2]
```diff
@@ -402,3 +408,74 @@ def dispatch(self, request, *args, **kwargs):
     def get_success_url(self):
         """Show confirmation."""
         return reverse('address-confirm')
+
+
+IMAGE_FILE_TYPES = ['png', 'jpg', 'jpeg', 'gif']
+
+
+class ProfileView(LoginRequiredMixin, DetailView, FormView):
```
Comment (SillyInventor) [inline project_share/views.py]: Please make a separate profileUpdate view

File: project_share/templates/project_share/user_detail.html [hunk 3]
```diff
@@ -11,6 +12,24 @@
         </h1>
     </div>
 
+    <h3>Update Profile:</h3>
```
Comment (SillyInventor) [inline project_share/templates/project_share/user_detail.html]: Please show avatar image... we may have to have some level of censorship for this at some point...

File: project_share/views.py (hunk 2 above)
Comment (ryaholm) [inline project_share/views.py]: done

File: project_share/views.py (hunk 1 above)
Comment (ryaholm) [inline project_share/views.py]: done

File: project_share/templates/project_share/user_detail.html (hunk 3 above)
Comment (ryaholm) [inline project_share/templates/project_share/user_detail.html]: done

**Pull request merged**